runtime/src/main/kotlin/kotlin/native/concurrent/MutableData.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package kotlin.native.concurrent
+
+import kotlin.native.internal.*
+import kotlinx.cinterop.*
+
+@SymbolName("Kotlin_Any_share")
+external private fun Any.share()
+
+@SymbolName("Kotlin_CPointer_CopyMemory")
+external private fun CopyMemory(to: COpaquePointer?, from: COpaquePointer?, count: Int)
+
+/**
+ * Mutable concurrently accessible data buffer. Could be accessed from several workers simulteniously.
+ */
+@Frozen
+public class MutableData {
```

Comment:
Should it be `@Experimental`? I am not sure this API is going to be stable in a nearest future

Reply:
Everything in kotlin.native.concurrent is experimental, no need to clutter it with uneeded annotations.

Reply:
`@Experimental` annotations are to make consumer opt in for experimental api, so they are surely not unneeded.

Reply:
As I mentioned, we do not expect much stability in kotlin.native.concurrent.